client/src/redux/reducers/datafiles.reducers.js

```diff
@@ -370,12 +370,14 @@ export function files(state = initialFilesState, action) {
         }
       };
     case 'STORE_SELECTOR_REF':
-      return {
-        ...state,
-        refs: {
-          ...state.refs,
-          FileSelector: action.payload
-        }
+      if(action.payload.props.isOpen){
```

Comment:
I think this will never be hit actually, since the modal will never be `isOpen` when the modal component initially renders in the `AppForm.js` - which would be when `isOpen === false`

Reply:
Yes, although, I don't know that we want this section to be triggered on initial render. My thinking is that although the dispatch here is `STORE_SELECTOR_REF` it's referencing the same DataFilesSelectModal that would be open if a user were navigating to a storage system. If while navigating to a file system (using DataFilesSelectModal) a user needed to push keys for that system the user would be presented with a push-keys modal over the currently open DataFilesSelectModal modal, and after pushing keys, the user would still be left with an open DataFilesSelectModal that now needs to be re-rendered.

Reply:
I mean that I don't think it would be ever hit, since the dispatch in `DataFilesSelectModal.js` takes place on initial render (when the modal is closed), and never again

Reply:
Oh, I thought it was triggered after a  successful key push which caused the DataFilesSelectModal to render after returning from a successful push

Reply:
Yeah it would re-render the component on refresh, but still closed, so I think it would still skip this block

Reply:
I have just looked at the code changes but haven't checked this out and tested yet.  but can we look into the `systems.sagas.js`. It feels like we should confirm the logic in the saga as well. This regression is related to `Bug/FP-642: Push key modal from select modal (#219)` so its possible that we didn't test missing execution system keys (and only system storage keys) when developing/testing/reviewing that PR.

Reply:
Ok, looks like we can dig a but more into the details, but maybe for the purposes of fixing this but we can just get rid of the dispatch causing the data files modal to open after push-keys. I can go ahead and pull this change and just remove the below, that should fix the bug and we can come back around to add the files-modal refresh. sound good?


`  useEffect(() => {
    dispatch({
      type: 'STORE_SELECTOR_REF',
      payload: selectRef.current
    });
  }, [selectRef]);`

Reply:
Sounds good to me, we'll follow up in a separate task

Reply:
Good idea 👍

Reply:
thanks, it's the fallback plan Sal and I had agreed to if we couldn't find an easy way to neatly wrap up a more complete solution

Should be updated now